internal/k8s/k8s.go

```diff
@@ -444,7 +446,7 @@ func (c *Client) Run(stopCh <-chan struct{}) error {
 		st := c.sync(key)
 		switch st {
 		case SyncStateSuccess:
-		case SyncStateErrorNoRetry:
+		case SyncStateErrorNoRetry, SyncStateDeleteSuccess:
```

Comment:
The previous version was returning SyncStateSuccess, so I'd expect `c.queue.Forget` to be called even in that scenario.
Can you explain why this version works and the previous one wasn't?

Reply:
Ok, cleared offline on the fact that go has no cascading cases.
Now the question is, why do we want this only for successful deletes and not for all `SyncStateSuccess` ?

I think it's fair to say that if we process an element successfully, we can forget about it. Also, with fallthrough you can enable cascading cases: https://stackoverflow.com/questions/40821855/do-go-switch-cases-fallthrough-or-not